src/process/process.cr

```diff
@@ -37,6 +38,19 @@ module Process
     LibC.getpid()
   end
 
+  def self.getpgid(pid : Int32)
+    gpid = LibC.getpgid(pid)
+    if gpid.is_a?(Int32) && gpid.to_i >= 0
```

Comment:
I think you're confusing things here, `fun getpgid(pid : Int32) : Int32` guarantees that the call returns an Int32, `gpid` is a local variable, it can't be changed by an external thread, so this should be `if gpid > 0`.